src/coreclr/interpreter/compiler.cpp

```diff
@@ -2262,9 +2651,54 @@ int InterpCompiler::GenerateCode(CORINFO_METHOD_INFO* methodInfo)
                     // We will just skip all instructions instead, since it doesn't seem that problematic.
                 }
             }
-            if (!m_pCBB->pNextBB)
-                m_pCBB->pNextBB = pNewBB;
+
+            InterpBasicBlock *pPrevBB = m_pCBB;
```

Comment:
It would be nice if this chunk could be extracted to a separate method. This bblock handling prologue is getting out of hand 😢

Reply:
Makes sense. I'll do it.